vali_objects/position.py

```diff
@@ -519,19 +519,24 @@ def _clamp_and_validate_leverage(self, order) -> bool:
         is_first_order = len(self.orders) == 0
         proposed_leverage = self.net_leverage + order.leverage
         min_position_leverage, max_position_leverage = leverage_utils.get_position_leverage_bounds(self.trade_pair, order.processed_ms)
-        if abs(proposed_leverage) > max_position_leverage:
-            if is_first_order or abs(proposed_leverage) >= abs(self.net_leverage):
-                order.leverage = max(0.0, max_position_leverage - abs(self.net_leverage))
-                if order.order_type == OrderType.SHORT:
-                    order.leverage *= -1
-                should_ignore_order = order.leverage == 0
-            else:
-                pass#  We are getting the leverage closer to the new boundary (decrease) so allow it
-        elif abs(proposed_leverage) < min_position_leverage:
-            if is_first_order or abs(proposed_leverage) < abs(self.net_leverage):
-                raise ValueError(f'Attempted to set position leverage below min_position_leverage {min_position_leverage}')
-            else:
-                pass  # We are trying to increase the leverage here so let it happen
+        # we only need to worry about clamping if the sign of the position leverage remains the same i.e. position does not flip and close
```

Comment:
if they trying to make a position's leverage really small, we should still block that. that can happen if the order leverage and the position leverage are different directions.

Reply:
Maybe make a explicit check for implicit close and then bypass all further checks.

Reply:
we could also just force close a position if they try taking the position leverage below the minimum position leverage

Reply:
This change doesn't affect the case where a miner attempts to make position leverage really small e.g. long 5 -> short 4.999.

It adds an explicit check for implicit close with `self.net_leverage * proposed_leverage > 0`. That makes sure that the sign of the position leverage remains the same.